python/social_bot/tasks.py

```diff
@@ -422,8 +427,12 @@ def _move_goal_impl(self, goal, agent_loc, agent_dir):
                    random.random() * range - range / 2, 0)
 
             self._initial_dist = np.linalg.norm(loc - agent_loc)
-            if self._initial_dist > self._success_distance_thresh:
+            if self._initial_dist > self._success_distance_thresh and (
+                attempts > 10000 or
+                (abs(loc[0]) < 5 and abs(loc[1]) < 5)  # within walls
```

Comment:
We can remove the default grey wall in the play ground world file, and insert them according to maximal random range. 
The wall should be lower, and adjustable in length. We can add a customized model for that.

Reply:
Very good point.  However, I'm hesitant to change the env as it makes old experiments ran on old env hard to compare with new runs on the new env. 
 I've added a TODO, and let's do this if absolutely needed?

Reply:
Not absolutely needed. Just a suggestion. I'll see if there is any elegant way to implement a wall model that is adjustable in length and height latter.

Reply:
Thanks Jiangtao.  It would be nice to have adjustable models, so we can control behavior using python.  We can also procedurally generate mazes etc..